clusterman/autoscaler/autoscaler.py

```diff
@@ -304,9 +305,9 @@ def _compute_target_capacity(self, resource_request: SignalResourceRequest) -> f
         elif non_orphan_fulfilled_capacity == 0:
             # Entering the main body of this method with non_orphan_fulfilled_capacity = 0 guarantees that
             # new_target_capacity will be 0, which we do not want (since the resource request is non-zero)
+            current_target_capacity += 1
```

Comment:
are we ok with potentially over-scaling to start with? there are some pools that run the autoscaler on a period < how long it takes nodes to come online - this change would lead us to keep adding more capacity until that first node joins the cluster, right? (even if everything is ok)

Reply:
Great point, I didn't know about frequency of calling method. We need to go with other way. Thank you @nemacysts !

Reply:
I agree with @nemacysts  current logic  may cause issues for some cluster.
I am thinking,  instead of this approach can we have timeout  which can recycle orphan node  if they are not joining cluster within some time limit as it will prevent us over-scaling also solves problem for waiting orphan node indefinitely.  thoughts ?

Reply:
@ankit864 Cluster will not start running any workload while this "timeout".
Yes, waiting for "timeout" is better than "indefinitely". But is it acceptable?
Do we know approximately "instance joining time" ?

Reply:
@ilkinmammadzada  I think the approach @ankit864 mentioned seems to be better suited since that would let us solve a few different issues regarding the ORPHAN node. The Key method to solve this problem is to force clusterman to launch new nodes even if nodes are stuck in the ORPHAN state. While Increasing the current_capacity can help us achieve this, we can achieve the same by deleting the nodes that are stuck in the ORPHAN state for a long time ( Deleting nodes would call a mismatch in target capacity and full-field capacity which would force clusterman to launch new nodes).

Reply:
I think we can use a default timeout ( around 30 mins). As a starting point, I think we can use the same config for all pools ( since 30 mins are plenty of time to join a cluster)